### ISSUE TITLE
Does not check if objective parameter is valid [Python]

### ISSUE THREAD

**Issue by cbonilla20** [dmlc/xgboost#387]
Testing the XGB with params, I put incorrectly the objective param

objective="reg:logitic"

The result was a run with an output of same predict for each case. This should check if the parameter was passing correctly

**Comment by tqchen** [comment]
Please provide a snippet to reproduce the problem.

**Comment by cbonilla20** [comment]
kf = cross_validation.KFold(len(y),n_folds=10, shuffle=True, indices=False)
actuals = y.copy()
predictions = y.copy()
# setup parameters for xgboost
param = {}
param['objective'] = 'reg:logitic'
param['silent'] = 1
param['nthread'] = 2
xgb_model = xgb.XGBRegressor(param)
for train_index, test_index in kf:
    xgb_model.fit(X[train_index],y[train_index])
    predictions[test_index] = xgb_model.predict(X[test_index])
    actuals[test_index] = y[test_index]
    print(actuals[test_index], predictions[test_index])

**Comment by tqchen** [comment]
I guess this have things to do with the way you pass into the XGBRegressor. Maybe you want to pass things in as arguments. Since XGBRegressor do not expect a dict as argument.

Or you can do  ``` xgb.XGBRegressor(**param)  ``` instead

**Comment by cbonilla20** [comment]
You are right about it. as dict argument respond as originally I placed on this issue. But then, the issue is that, not solving the params as dict argument or directly launch exception for that. Just a suggestion. Regards, and thank you for answer me.

**Comment by tqchen** [comment]
This is actually a good suggestion. Do you want to create a patch on this? Thanks!
